sale_order_project/models/sale.py

```diff
@@ -22,48 +22,45 @@
 #
 ###############################################################################
 
-from openerp.osv import fields, orm
+from openerp import api, fields, models
 from datetime import date
 
 
-class sale_order(orm.Model):
+class SaleOrder(models.Model):
     _inherit = "sale.order"
 
-    _columns = {
-        'related_project_id': fields.many2one(
-            'project.project',
-            'Project',
-            readonly=True,
-            states={'draft': [('readonly', False)]}
-        ),
-    }
+    @api.one
+    @api.depends('project_id')
+    def _compute_related_project_id(self):
+        if self.project_id.use_tasks:
+            related_project_id = self.env['project.project'].search(
```

Comment:
Make directly:
```
self.related_project_id = self.env['project.project'].search([('analytic_account_id', '=', self.project_id.id)], limit=1)[:1]
```
without making related_project_id = False, because this is set by default in ORM